wagtailio/project_styleguide/templates/patterns/components/theme-toggle/theme-toggle.html

```diff
@@ -0,0 +1,11 @@
+<div class="theme-toggle">
+    <noscript>
+        <p>{{ _('JavaScript is required to toggle light/dark mode.') }}.</p>
+    </noscript>
+    <span class="theme-toggle__label">Dark</span>
+    <button id="wagtail-theme" class="theme-toggle__button" type="button" aria-label="Toggle between dark and light theme">
+        <span class="dark-only u-sr-only">Light mode</span>
+        <span class="light-only u-sr-only">Dark mode</span>
```

Comment:
I don't think these `u-sr-only` spans will be announced due to the `aria-label` on the button. It might be better to just have the aria-label that updates when clicked and gives context to the user - you could add something like the following in `blocking.js`

```
    if (event) {
        const toggleButton = document.getElementById('wagtail-theme');
        if (applyMode === 'light') {
            toggleButton.setAttribute('aria-label', 'Change to Dark Theme');
        } else {
            toggleButton.setAttribute('aria-label', 'Change to Light Theme');
        }
    }
```

You'd also need to add in the JS to detect on page load as well.